src/Optimization/hiopAlgFilterIPM.cpp

```diff
@@ -928,12 +886,56 @@ void hiopAlgFilterIPMBase::displayTerminationMsg()
 ///////////////////////////////////////////////////////////////////////////////////////////////////
 // hiopAlgFilterIPMQuasiNewton
 ///////////////////////////////////////////////////////////////////////////////////////////////////
-hiopAlgFilterIPMQuasiNewton::hiopAlgFilterIPMQuasiNewton(hiopNlpDenseConstraints* nlp_,
+hiopAlgFilterIPMQuasiNewton::hiopAlgFilterIPMQuasiNewton(hiopNlpDenseConstraints* nlp_in,
                                                          const bool within_FR)
-  : hiopAlgFilterIPMBase(nlp_, within_FR)
+  : hiopAlgFilterIPMBase(nlp_in, within_FR)
 {
-  nlpdc = nlp_;
-  //_Hess = new hiopHessianLowRank(nlpdc, nlpdc->options->GetInteger("secant_memory_len"));
+  nlpdc = nlp_in;
+  reload_options();
+
+  it_curr = new hiopIterate(nlp);
+  it_trial= it_curr->alloc_clone();
+  dir     = it_curr->alloc_clone();
+
+  if(nlp->options->GetString("KKTLinsys")=="full") {
+    it_curr->selectPattern();
+    it_trial->selectPattern();
+    dir->selectPattern();
+  }
+
+  logbar = new hiopLogBarProblem(nlp);
+
+  _f_nlp = _f_log = 0;
+  _c = nlp->alloc_dual_eq_vec();
+  _d = nlp->alloc_dual_ineq_vec();
+
+  _grad_f  = nlp->alloc_primal_vec();
+  _Jac_c   = nlp->alloc_Jac_c();
+  _Jac_d   = nlp->alloc_Jac_d();
+
+  _f_nlp_trial = _f_log_trial = 0;
+  _c_trial = nlp->alloc_dual_eq_vec();
+  _d_trial = nlp->alloc_dual_ineq_vec();
+
+  _grad_f_trial  = nlp->alloc_primal_vec();
+  _Jac_c_trial   = nlp->alloc_Jac_c();
+  _Jac_d_trial   = nlp->alloc_Jac_d();
+
+  _Hess_Lagr = nlp->alloc_Hess_Lagr();
+
+  resid = new hiopResidual(nlp);
+  resid_trial = new hiopResidual(nlp);
+
+  //parameter based initialization
+  if(duals_update_type==0) {
+    dualsUpdate_ = nlp->alloc_duals_lsq_updater();
+  } else if(duals_update_type==1) {
+    dualsUpdate_ = new hiopDualsNewtonLinearUpdate(nlp);
+  } else {
+    assert(false && "duals_update_type has an unrecognized value");
+  }
+
+  resetSolverStatus();
 }
```

Comment:
Many variables are initialized in the derived classes now, not in the base class `hiopAlgFilterIPMBase` anymore.
Any particular reason to duplicate these codes?

Reply:
yes, because `reload_options` called by the constructor is virtual and needs to be called in the child classes. But I will add a couple of methods to hold the common code.